lib/kubernetes-deploy/resource_discovery.rb

```diff
@@ -10,7 +10,7 @@ def initialize(namespace:, context:, logger:, namespace_tags:)
     end
 
     def crds(sync_mediator)
-      sync_mediator.get_all(CustomResourceDefinition.kind).map do |r_def|
+      @crds ||= sync_mediator.get_all(CustomResourceDefinition.kind).map do |r_def|
```

Comment:
Should we save the `ResourceDiscovery` instance into a ivar is deploy_task.rb?

Reply:
This PR does do that (the method is `cluster_resource_discoverer`)